group_project/group_activity.py

```diff
@@ -1,3 +1,75 @@
+"""
+This module parses Group project XML into a tree of domain-specific objects.
+Basically it looks like the following
+
+GroupActivity (.)
+- ActivityComponent (./projectcomponent)
+-- ActivitySection (//section - different paths)
+--- ActivityQuestion (//section/question)
+---- <no dedicated class, etree.Element is used> (//section/question/answer)
+--- ActivityAssessment (//section/assessment)
+---- <no dedicated class, etree.Element is used> (//section/assessment/answer)
+
+GroupActivity (paths relative to root element)
+- resources: [DottableDict(title, description, location)] - ./resources/document
+- grading_criteria: [DottableDict(title, description, location)] - ./resources/document[@grading_criteria="true"]     # attribute filter implicit
+- milestone_dates: {str: datetime.date} - ./dates/milestone
+- submissions: [DottableDict(id, title, description, location?)] - ./submissions/document
+- grade_questions: [ActivityQuestion] - //section/question
```

Comment:
* Resources: You're right - individual resources children would get added to each activity block. Although the ResourcesList XBlock (contained by the top-level Navigator XBlock) could walk the tree of group work xblocks (each representing an activity) to populate the list of resources for each activity. This way it would allow authors to include resources at any level (for example in individual stages), which can avoid cluttering the interface in Studio at the activity level. This would also make the Resources and ResourcesList more generic, and allow easier reuse outside of the group work XBlock.
* milestones_dates: Yes, that seem like an artifact - deadlines should really be handled by the type of Stage XBlocks which have deadlines, there isn't really a reason to make them their own individual XBlock. If we need to have name references, that could be through names given to individual stages - it seem to always have only one deadline possible per stage, correct?
* Same remark as for resources - this could be part of individual stages, as if I understand correctly an upload is needed for completing a given stage, right? This would avoid cluttering the activity level, and having to link to a particular upload from a particular stage, which would become obvious from its position in the tree.